### PR TITLE
Type support

### PR DESCRIPTION
**Pull request by vaind**
* cleanup
* StringVector
* BytesVector

### CODE REVIEW

**Comment by vaind** [comment]
@greenrobot-team please have a look. I'll add queries later but none of the existing code should change

**Comment by greenrobot-team** [comment]
First Code analysis / lint failed due to `At the time of the analysis https://objectbox.io was unreachable.`, nothing to see here, moving along.

File: test/query_test.dart [hunk 1]
```diff
@@ -198,6 +198,24 @@ void main() {
...
+
+    final prop = TestEntity_.tStrings;
+
+    final qs0 = box.query(prop.contains('bar')).build();
+    final qs1 = box.query(prop.contains('ar')).build();
+    final qs2 = box.query(prop.contains('foo')).build();
+
+    expect(qs0.count(), 1);
+    expect(qs1.count(), 0);
+    expect(qs2.count(), 2);
```
Comment (greenrobot-team) [inline test/query_test.dart:214]: I find this hard to read. Maybe something like this?
```suggestion
    final qs0 = box.query(prop.contains('bar')).build();
    expect(qs0.count(), 1);
    
    final qs1 = box.query(prop.contains('ar')).build();
    expect(qs1.count(), 0);
    
    final qs2 = box.query(prop.contains('foo')).build();
    expect(qs2.count(), 2);
```

**Review by greenrobot-team — CHANGES_REQUESTED** [review]
Some minor stuff, otherwise LGTM 👍

File: generator/lib/src/entity_resolver.dart [hunk 2]
```diff
@@ -135,9 +139,23 @@ class EntityResolver extends Builder {
           // dart: 8 bytes
           // ob: 8 bytes
           fieldType = OBXPropertyType.Double;
+        } else if (fieldTypeDart.isDartCoreList &&
+            listItemType(fieldTypeDart).isDartCoreString) {
+          // List<String>
+          fieldType = OBXPropertyType.StringVector;
+        } else if (fieldTypeDart.element.name == 'Int8List') {
```
[the remark this answers is not in the document]
Comment (vaind) [reply generator/lib/src/entity_resolver.dart:146]: `.toString()` is different and more likely to change I'd say. 

I've found out about that `.element.name` matching when looking at the dart core implementation, .e.g.:
```dart
@override
  bool get isDartCoreInt {
    return element.name == "int" && element.library.isDartCore;
  }
```

File: lib/src/annotations.dart [hunk 3]
```diff
@@ -13,13 +13,69 @@ class Entity {
...
+  final PropertyType /*?*/ type;
 
   const Property({this.type, this.flag, this.uid});
 }
 
+// Specify ObjectBox property type explicitly
+enum PropertyType {
+  /// dart type=bool, size: 1-byte/8-bits
+  /// no need to specify explicitly, just use `bool`
+  // bool,
```
[the remark this answers is not in the document]
Comment (vaind) [reply lib/src/annotations.dart:28]: fixed all of those

File: generator/lib/src/entity_resolver.dart [hunk 4]
```diff
@@ -233,16 +250,41 @@ class EntityResolver extends Builder {
...
+        .fields
+        .where((f) => f.isEnumConstant)
+        .toList();
+
+    // Find the index of the matching enum constant.
+    for (var i = 0; i < enumValues.length; i++) {
+      if (enumValues[i].computeConstantValue() == typeField) {
+        return propertyTypeToOBXPropertyType(obx.PropertyType.values[i]);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (vaind) [reply generator/lib/src/entity_resolver.dart:280]: done

File: generator/lib/src/entity_resolver.dart [hunk 5]
```diff
@@ -135,9 +139,23 @@ class EntityResolver extends Builder {
...
+            listItemType(fieldTypeDart).isDartCoreString) {
+          // List<String>
+          fieldType = OBXPropertyType.StringVector;
+        } else if (fieldTypeDart.element.name == 'Int8List') {
+          fieldType = OBXPropertyType.ByteVector;
+          dartFieldType =
+              fieldTypeDart.element.name; // needed for code generation
+        } else if (fieldTypeDart.element.name == 'Uint8List') {
+          fieldType = OBXPropertyType.ByteVector;
+          // TODO check if UNSIGNED also applies to byte-vector in the core
```
[the remark this answers is not in the document]
Comment (vaind) [reply generator/lib/src/entity_resolver.dart:152]: not needed, byte vectors seem to be only unsigned in the core

File: test/query_test.dart (hunk 1 above)
Comment (vaind) [reply test/query_test.dart:214]: applied the suggestion

**Review by greenrobot-team — APPROVED** [review]
LGTM!

**Pull request merged**